design.md

```diff
@@ -0,0 +1,49 @@
+
```

Comment:
MIght be prudent to break this out into another directory. Designs should be spread across different files as the collection should grow in the long term

Reply:
Good idea. Going to merge for now and lets discuss how to document designs synchronously.